hassio/hassio-main.html

```diff
@@ -44,58 +51,77 @@
       ></hass-loading-screen>
     </template>
     <template is='dom-if' if='[[loaded]]'>
-      <iron-pages
-        selected='[[_routeData.page]]'
-        attr-for-selected='page-name'
-        fallback-selection='not-found'
-        selected-attribute='visible'
-      >
-        <hassio-addon-view
-          page-name='addon'
-          route='[[route]]'
-          hass='[[hass]]'
-        ></hassio-addon-view>
-
-        <hassio-addon-store
-          page-name='store'
-          hass='[[hass]]'
-        ></hassio-addon-store>
-
-        <hassio-supervisor
-          page-name='supervisor'
-          hass='[[hass]]'
-          supervisor-info='[[supervisorInfo]]'
-        ></hassio-supervisor>
-
-        <hassio-dashboard
-          page-name='dashboard'
-          hass='[[hass]]'
-          narrow='[[narrow]]'
-          show-menu='[[showMenu]]'
-          supervisor-info='[[supervisorInfo]]'
-          hass-info='[[hassInfo]]'
-        ></hassio-dashboard>
-
-        <hassio-advanced
-          page-name='advanced'
-          hass='[[hass]]'
-          supervisor-info='[[supervisorInfo]]'
-          host-info='[[hostInfo]]'
-          hass-info='[[hassInfo]]'
-        ></hassio-advanced>
-
-        <hassio-snapshots
-          page-name='snapshot'
-          hass='[[hass]]'
-          installed-addons='[[supervisorInfo.addons]]'
-        ></hassio-snapshots>
-
-        <hass-error-screen
-          page-name='not-found'
-          error='Page not found.'
-          title='Hass.io'
-        ></hass-error-screen>
-      </iron-pages>
+      <app-header-layout has-scrolling-region>
+        <app-header fixed slot='header'>
+          <app-toolbar>            
+            <ha-menu-button narrow='[[narrow]]' show-menu='[[showMenu]]'></ha-menu-button>
+            <paper-icon-button
+              icon='mdi:arrow-left'
+              on-tap='backTapped'
+              hidden='[[computeHideBack(mainTitle)]]'></paper-icon-button>
+            <div main-title>[[mainTitle]]</div>
+          </app-toolbar>
+          <paper-tabs
+            scrollable
+            selected='[[routeData.page]]'
+            attr-for-selected='page-name'
+            on-iron-activate='handlePageSelected'
+          >
+            <paper-tab page-name='dashboard'>Dashboard</paper-tab>
+            <paper-tab page-name='store'>Add-on store</paper-tab>
+            <paper-tab page-name='snapshots'>Snapshots</paper-tab>
+            <paper-tab page-name='advanced'>Advanced</paper-tab>
+          </paper-tabs>
+        </app-header>
+        <iron-pages
```

Comment:
You're coding yourself into a corner here with this information architecture. From a tabbed interface, how will you handle navigation to a page deeper in the information structure? Or will you use dialogs for this? That will only work once. Then you have a dialog open, what then?

hass.io root
 - dashboard
    - details installed add-on
 - addon store
    - details add-on
 - Snapshots
    - details Snapshot
 - Advanced
    - Supervisor card
          - Supervisor logs?
    - Host

That's why I normally go with replacing the whole page and show a back button in top right. That way you can deep as you want.

Reply:
First we used routeData for snapshot details but cause a) we only show a small amount of information and b) why should someone link to a snapshot?, we switched to paperdialog's for this.
For add-on details pages we show a back-button and change the header name to "Add-on details"

add-on is installed:
![ai](https://user-images.githubusercontent.com/11984118/36060956-6e187a72-0e53-11e8-9397-27d9d60352bf.png)

add-on not installed:
![ani](https://user-images.githubusercontent.com/11984118/36060957-72015320-0e53-11e8-98ef-508d5519b9e2.png)

^maybe you criticize here, that add-on is not listed in the nav tabs 

I don't see where we would use more than 1-2 subpages so navigation structure should be no problem?

hass.io root
 - dashboard `/hassio/dashboard`
    - details installed add-on `/hassio/addon/core_configurator`, back button enabled
 - addon store `/hassio/store`
    - details add-on `/hassio/addon/core_configurator`, back button enabled
 - Snapshots `/hassio/snapshots`
    - details Snapshot `-> paperdialog`
 - Advanced `/hassio/advanced`
    - Supervisor card
          - Supervisor logs? `/hassio/supervisor`, no back button yet, could add it
    - Host `?`

Why tab navigation:
-we don't use thise top right menu/ buttons anywhere else (only maybe for voice input)
-makes pages like store and snapshot more accessible (or why do we have to use red arrows to point to the add-on store in our docs?)

Reply:
The add-on details and then still showing the tab nav is very weird. I've never seen that in any app. What happens when I click on a tab? At least the title needs to stay the same.

Icons on top right is very common in material design. If it is not obvious enough, we could add a card with a link to the main dashboard.